arenadata/arenadata_helper.go

```diff
@@ -0,0 +1,41 @@
+package arenadata
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/greenplum-db/gp-common-go-libs/gplog"
+	"github.com/pkg/errors"
+)
+
+func EnsureAdVersionCompatibility(backupVersion string, restoreVersion string) {
+	regex := *regexp.MustCompile(`_arenadata(\d+)`)
+	adBackup := regex.FindAllStringSubmatch(backupVersion, -1)[0][1]
+	adRestore := regex.FindAllStringSubmatch(restoreVersion, -1)[0][1]
+	if adRestore < adBackup {
```

Comment:
Are you sure that we compare versions as integers, not strings?

Reply:
oops

Reply:
Maybe add/replace a test case that checks such error?

Reply:
done
https://github.com/arenadata/gpbackup/blob/3d48647502ae1e961d736748417e2f44b526880c/report/report_test.go#L395-L401